src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts

```diff
@@ -580,6 +580,16 @@ const terminalConfiguration: IConfigurationNode = {
 			type: 'boolean',
 			default: false
 		},
+		[TerminalSettingId.ShouldUseSystemd]: {
+			markdownDescription: localize('terminal.integrated.systemd.shouldUse', 'Controls whether to use systemd for terminals where possible'),
+			type: 'boolean',
+			default: false
+		},
+		[TerminalSettingId.SystemdSliceSuffix]: {
+			markdownDescription: localize('terminal.integrated.systemd.sliceSuffix', 'Name of systemd slice suffix for terminals.  This will always be prefixed with `vscode_`'),
+			type: 'string',
+			default: 'terminal'
+		}
```

Comment:
I actually added this for this diff (we originally didn't have it) since I didn't want to try to upstream our hardcoded values.  How would we do a "hidden setting"?  Is it the same as configs with a different flag or is there another mechanism?  This is definitely niche but someone who actually uses it would want to configure the slice they're using.  Definitely open to ideas to change this as appropriate.

Reply:
My concerns about this PR are the fact that I can't really maintain it as I don't use Linux very often and systemd is totally new to me, and the fact that these 2 settings that are applicable to < 1% of users will bloat the terminal settings more than they already are.

By hidden setting I mean just not registering them, but still accessing like normal in the config service. Another idea which is probably better as it handles the can't maintain problem above is for us to expose some hook(s) somewhere that you could easily apply a patch on top of, and then commit to them outside of major refactors.

Alternatively, could you just insert something [here](https://github.com/microsoft/vscode/blob/073705b1a98c16d9e4c9ad74bfa6cab2ed14bdba/src/vs/platform/terminal/node/terminalProcess.ts#L164) in a custom patch that manipulates the exe/args/env? That seems like a fairly easy to maintain patch on your end.

Reply:
OK.  It sounds like this may not be a good candidate for contribution then given your thoughts on the audience reach, which are totally valid.  Reverting the settings infra on our end makes this fork more manageable.  

We can close this PR then and we'll keep this as a fork.